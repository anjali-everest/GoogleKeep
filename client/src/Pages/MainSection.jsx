import React from "react"
import { useSelector } from "react-redux"
import {
    NotesList
} from './feature/notes/NotesList';
import { AddNoteForm } from "./feature/notes/AddNoteForm";
import { EditNoteForm } from "./feature/notes/EditNoteForm";

const MainSection = () => {
    const isEditingNote = useSelector(state => state.notes.isEditingNote)

    return (
        <>
            <AddNoteForm />
            <NotesList />
            {isEditingNote && <EditNoteForm />}
        </>
    )
}

export default MainSection;