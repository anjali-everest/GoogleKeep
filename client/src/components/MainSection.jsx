import React from "react";
import { useSelector } from "react-redux";
import { NotesList } from "../features/notes/components/NotesList";
import { AddNoteForm } from "../features/notes/components/AddNoteForm";
import { EditNoteForm } from "../features/notes/components/EditNoteForm";
import { selectIsEditingNote } from "../features/notes/notesSlice";

const MainSection = () => {
  const isEditingNote = useSelector(selectIsEditingNote);

  return (
    <div className="main-section">
      <AddNoteForm />
      <NotesList />
      {isEditingNote && <EditNoteForm />}
    </div>
  );
};

export default MainSection;
