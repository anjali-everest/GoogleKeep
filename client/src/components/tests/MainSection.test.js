import React from "react";
import { mount } from "enzyme";
import MainSection from "../MainSection";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
const mockStore = configureMockStore();

describe("MainSection tests", () => {
  const store = mockStore({
    notes: [],
    status: "idle",
    error: null,
    editingNoteId: null,
    isEditingNote: false,
    deletingNoteId: null,
  });

  it("Should render without crashing", () => {
    mount(
      <Provider store={store}>
        <MainSection />
      </Provider>
    );
  });
  it("should match the MainSection snapshot", function () {
    const mainSection = mount(
      <Provider store={store}>
        <MainSection />
      </Provider>
    );
    expect(mainSection).toMatchSnapshot();
  });
});
