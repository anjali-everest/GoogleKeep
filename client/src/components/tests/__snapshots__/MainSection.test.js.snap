// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSection tests should match the MainSection snapshot 1`] = `ReactWrapper {}`;
